tests/e2e/vsan_stretched_cluster.go

```diff
@@ -363,4 +376,482 @@ var _ = ginkgo.Describe("[vsan-stretch-vanilla] vsan stretched cluster tests", f
 		scaleDownNDeleteStsDeploymentsInNamespace(ctx, client, namespace)
 	})
 
+	/*
+	   Statefulset scale up/down while primary site goes down
+	   Steps:
+	   1.  Configure a vanilla multi-master K8s cluster with inter and intra site replication
+	   2.  Create two statefulset with replica count 1(sts1) and 5(sts2) respectively with two PVCs each
+	       and wait for all replicas to be running
+	   3.  Change replica count of sts1 and sts2 to 3
+	   4.  Bring down primary site
+	   5.  Verify that the VMs on the primary site are started up on the other esx servers in the secondary site
+	   6.  Verify there were no issue with replica scale up/down and verify pod entry in CNS volumemetadata for the volumes associated
+	       with the PVC used by statefulsets are updated
+	   7.  change replica count of sts1 to 5 a sts2 to 1 and verify they are successful
+	   8.  Delete statefulsets and its pvcs created in step 2
+	   9.  Bring primary site up and wait for testbed to be back to normal
+	*/
+	ginkgo.It("Statefulset scale up/down while primary site goes down", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		ginkgo.By("Creating StorageClass for Statefulset")
+		// decide which test setup is available to run
+		ginkgo.By("CNS_TEST: Running for vanilla k8s setup")
+		scParameters = map[string]string{}
+		scParameters["StoragePolicyName"] = storagePolicyName
+		storageClassName = "nginx-sc-default"
+
+		scSpec := getVSphereStorageClassSpec(storageClassName, scParameters, nil, "", "", false)
+		sc, err := client.StorageV1().StorageClasses().Create(ctx, scSpec, metav1.CreateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, sc.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Creating service")
+		service := CreateService(namespace, client)
+		defer func() {
+			deleteService(namespace, client, service)
+		}()
+		statefulset := GetStatefulSetFromManifest(namespace)
```

Comment:
rename `statefulset` to `statefulset1` and `statefulset1` to `statefulset2`, this will help to correlate better with the steps mentioned above